src/main/java/handler/BalanceOperationHandler.java

```diff
@@ -0,0 +1,17 @@
+package handler;
+
+import java.util.Map;
+import model.FruitTransaction;
+
+public class BalanceOperationHandler implements OperationHandler {
+    private final Map<String, Integer> storage;
+
+    public BalanceOperationHandler(Map<String, Integer> storage) {
```

Comment:
Remember to follow the encapsulation principle. The storage map should be private to ensure that it is only accessed and modified through controlled methods.